specification/redis/Cache.Management/RedisResource.tsp

```diff
@@ -0,0 +1,228 @@
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+
+namespace Microsoft.Cache;
+/**
+ * A single Redis item in List or Get Operation.
+ */
+model RedisResource
+  is Azure.ResourceManager.TrackedResource<RedisProperties, false> {
+  ...ResourceNameParameter<
+    Resource = RedisResource,
+    KeyName = "cacheName",
+    SegmentName = "redis",
+    NamePattern = "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$"
+  >;
+  ...Azure.ResourceManager.AvailabilityZonesProperty;
+
+  /**
+   * The identity of the resource.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  identity?: Azure.ResourceManager.CommonTypes.ManagedServiceIdentity;
+}
+
+@armResourceOperations
+interface RedisResourcesOps
+  extends Azure.ResourceManager.Legacy.LegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+        ...Azure.ResourceManager.Legacy.Provider,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = RedisResource,
+        KeyName = "name",
+        SegmentName = "redis",
+        NamePattern = ""
+      >>
+    > {}
+
+@armResourceOperations
+interface RedisResources {
+  /**
+   * Gets a Redis cache (resource description).
+   */
+  get is RedisResourcesOps.Read<RedisResource>;
+
+  /**
+   * Create or replace (overwrite/recreate, with potential downtime) an existing Redis cache.
+   */
+  @put
+  create is Azure.ResourceManager.Legacy.UpdateOperation<
+    {
+      ...ApiVersionParameter;
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...Azure.ResourceManager.Legacy.Provider;
+
+      #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+      @segment("redis")
+      @path
+      name: string;
+    },
+    RedisCreateParameters,
+    RedisResource | ArmResourceCreatedResponse<
+      RedisResource,
+      LroHeaders = ArmLroLocationHeader<FinalResult = RedisResource>
+    >,
+    ErrorResponse
+  >;
+
+  /**
+   * Update an existing Redis cache.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/no-response-body" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @patch(#{ implicitOptionality: false })
+  update is RedisResourcesOps.CustomPatchAsync<
+    RedisResource,
+    PatchModel = RedisUpdateParameters,
+    Response = ArmResponse<RedisResource> | (ArmAcceptedLroResponse & {
+      @bodyRoot
+      _: RedisResource;
+    })
+  >;
+
+  /**
+   * Deletes a Redis cache.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  delete is RedisResourcesOps.DeleteWithoutOkAsync<
```

Comment:
Justification would be backCompat right?

Reply:
Do we just document these as back compat? Or do you wish to leave it as FIXME in case we improve the API in a future version?